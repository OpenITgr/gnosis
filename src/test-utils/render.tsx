/* eslint-disable import/export */
import React, { FC, ReactElement } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options?: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };
