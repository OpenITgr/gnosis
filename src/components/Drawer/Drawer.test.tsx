import React from "react";
import userEvent from "@testing-library/user-event";
import { Default as Drawer } from "./Drawer.stories";
import { screen, render } from "@test-utils/render";

describe("<Drawer/>", () => {
  const drawerEl = document.createElement("div");
  drawerEl.setAttribute("id", "drawerRoot");
  document.body.appendChild(drawerEl);

  it("renders correctly", () => {
    const mockFn = jest.fn();

    render(<Drawer onClose={mockFn} {...Drawer.args} />);

    userEvent.click(screen.getByRole("button"));

    const header = screen.getByText("Drawer title");
    const body = screen.getByText("Drawer content");
    const footer = screen.getByText("Close Drawer");
    const mask = screen.getByTestId("mask");
    const closeBtn = screen.getByLabelText("Close drawer");

    userEvent.click(mask);
    userEvent.click(closeBtn);

    expect(header).toHaveTextContent("Drawer title");
    expect(body).toHaveTextContent("Drawer content");
    expect(footer).toHaveTextContent("Close Drawer");
    expect(closeBtn).toBeInTheDocument();
    expect(mockFn).toHaveBeenCalledTimes(2);
  });

  it("matches snapshot with header, body and footer", () => {
    const mockFn = jest.fn();

    const { container } = render(<Drawer onClose={mockFn} {...Drawer.args} />);

    expect(container).toMatchSnapshot();
  });
});
