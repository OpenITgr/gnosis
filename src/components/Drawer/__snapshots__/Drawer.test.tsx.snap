// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer/> matches snapshot with close button 1`] = `<div />`;

exports[`<Drawer/> matches snapshot with header, body and footer 1`] = `
<div>
  <button
    class="solid css-18yafla-baseButton-baseButton-solidButton-solidButton-outlineButton-outlineButton-ghostButton-ghostButton-linkButton-linkButton-btnContainer-btnContainer"
  >
    <span
      class="btn-text"
    >
      Open Drawer
    </span>
  </button>
</div>
`;

exports[`<Drawer/> matches snapshot without header and footer 1`] = `<div />`;
